tokio/tests/io_async_fd.rs

```diff
@@ -947,12 +947,76 @@ async fn try_new() {
     assert!(Arc::ptr_eq(&original, &returned));
 }
 
+/// Regression test for issue #7563
+///
+/// Reproduces the bug where closing fd before dropping AsyncFd causes
+/// OS deregister to fail, preventing cleanup and leaking ScheduledIo objects.
 #[tokio::test]
-async fn try_with_interest() {
-    let original = Arc::new(InvalidSource);
+async fn memory_leak_when_fd_closed_before_drop() {
+    use std::os::unix::io::{AsRawFd, RawFd};
+    use std::sync::Arc;
+    use tokio::io::unix::AsyncFd;
```

Comment:
These are already imported in the beginning of the file. No need to import them again.